source/HailPanel.cpp

```diff
@@ -214,14 +214,12 @@ void HailPanel::Draw()
 
 	DrawList draw;
 	// If this is a ship, copy its swizzle, animation settings, etc.
+	// Also draw its fighters and weapon hardpoints.
 	if(ship)
-		draw.Add(Body(*ship, center, Point(), facing, zoom));
-	else
-		draw.Add(Body(sprite, center, Point(), facing, zoom));
-
-	// If hailing a ship, draw its turret sprites.
-	if(ship)
-		for(const Hardpoint &hardpoint : ship->Weapons())
+	{
+		bool hasFighters = ship->PositionFighters();
```

Comment:
For the sake of being explicit, perhaps rename this to "hasVisibleFighters." Although I do notice that this is the same variable name that Engine uses when drawing a ship sprite.

Speaking of which, a fair amount of this drawing code is rather similar to what Engine does when drawing the ship. Perhaps we could have some ship drawing function that both locations could share. Not something we need to do here, but something to keep in mind as a potential future change; it's not the end of the world to have similar looking code in two places.

(I've also realized that we're not drawing the hardpoints of the fighters, which could exist, in either location, so that's something else we could keep in mind.)

Reply:
Yeah, I just took the code from Engine and adjusted it to HailPanel context.

> (I've also realized that we're not drawing the hardpoints of the fighters, which could exist, in either location, so that's something else we could keep in mind.)

I tried to display them in Engine, but for some reason it didn't work.